### PR TITLE
Sol map event details

### PR DESCRIPTION
**Pull request by Sosol02**

### CODE REVIEW

File: app/src/main/java/com/github/onedirection/navigation/fragment/map/MapFragment.java [hunk 1]
```diff
@@ -62,6 +84,19 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
         super.onViewCreated(view, savedInstanceState);
     }
 
+    public void showBottomSheet() {
+        bottomSheetBehavior.setState(BottomSheetBehavior.STATE_COLLAPSED);
+    }
+
+    public void setBottomSheetEvent(Event event){
```
Comment (Flechman) [inline app/src/main/java/com/github/onedirection/navigation/fragment/map/MapFragment.java:91]: A little Objects.requireNonNull(event) would be nice

File: app/src/main/java/com/github/onedirection/navigation/fragment/map/MarkerSymbolManager.java [hunk 2]
```diff
@@ -1,53 +1,132 @@
...
 
+    public static final String LOG_TAG = "MarkerSymbolManager";
+
     private final SymbolManager symbolManager;
     private final List<Symbol> markers;
+    private final Map<Symbol, Event> eventMap = new HashMap<>();
+    private final MapFragment fragment;
 
-    public MarkerSymbolManager(SymbolManager symbolManager) {
+    public MarkerSymbolManager(SymbolManager symbolManager, MapFragment fragment) {
```
Comment (Flechman) [inline app/src/main/java/com/github/onedirection/navigation/fragment/map/MarkerSymbolManager.java:34]: Null checks missing here as well

**Review by Flechman — APPROVED** [review]
Great code Clean code

File: app/src/main/java/com/github/onedirection/navigation/fragment/map/MapFragment.java (hunk 1 above)
Comment (Sosol02) [reply app/src/main/java/com/github/onedirection/navigation/fragment/map/MapFragment.java:91]: The reasoning is that if event is null it crashes with a nullpointerexception anyways, but i can change that.

File: app/src/main/java/com/github/onedirection/navigation/fragment/map/MapFragment.java [hunk 3]
```diff
@@ -44,16 +58,25 @@ public View onCreateView(@NonNull LayoutInflater inflater,
...
 
             mapboxMap.addOnMapClickListener(point -> {
-                markerSymbolManager.removeAllMarker();
-                markerSymbolManager.addMarker(point);
+                if (clickSymbol != null)
+                    markerSymbolManager.removeMarker(clickSymbol);
+                clickSymbol = markerSymbolManager.addMarker(point);
                 return false;
-            });
+            }); // */
```
Comment (TheImpressiveDonut) [inline app/src/main/java/com/github/onedirection/navigation/fragment/map/MapFragment.java:68]: Forgot to remove useless commentary

File: app/src/main/java/com/github/onedirection/navigation/fragment/map/MapFragment.java [hunk 4]
```diff
@@ -96,6 +133,13 @@ private void initializeMarkerSymbolManager(@NonNull Style styleOnLoaded) {
...
         SymbolManager symbolManager = new SymbolManager(mapView, mapboxMap, styleOnLoaded);
-        this.markerSymbolManager = new MarkerSymbolManager(symbolManager);
+        this.markerSymbolManager = new MarkerSymbolManager(symbolManager, this);
+
+        /*
+        Event event = new Event(Id.generateRandom(), "Test event", "Paris",
+                ZonedDateTime.of(2021, 4, 2, 13, 42, 56, 0, ZoneId.systemDefault()),
+                ZonedDateTime.of(2021, 4, 2, 13, 58, 56, 0, ZoneId.systemDefault()));
+        markerSymbolManager.addGeocodedEventMarker(event).thenApply(symbol -> Log.d("Debug", "Symbol: " + symbol + " Map: " + markerSymbolManager.getEventMap()));
+        // */
```
Comment (TheImpressiveDonut) [inline app/src/main/java/com/github/onedirection/navigation/fragment/map/MapFragment.java:143]: Should we leave this comment?

File: app/src/main/res/layout/fragment_map.xml [hunk 5]
```diff
@@ -9,7 +9,67 @@
...
+            android:textSize="30sp" />
+
+        <TextView
+            android:id="@+id/fragment_map_event_time_start"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_margin="15dp"
+            app:layout_constraintTop_toBottomOf="@+id/fragment_map_event_name"
+            app:layout_constraintStart_toStartOf="parent"
+            android:text="Start time"
```
Comment (TheImpressiveDonut) [inline app/src/main/res/layout/fragment_map.xml:48]: Hardcoded string, put it in strings.xml  it is better

File: app/src/main/res/layout/fragment_map.xml [hunk 6]
```diff
@@ -9,7 +9,67 @@
...
+        android:background="@color/white">
+
+        <TextView
+            android:id="@+id/fragment_map_event_name"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_margin="15dp"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent"
+            android:text="Event name"
```
Comment (TheImpressiveDonut) [inline app/src/main/res/layout/fragment_map.xml:38]: Hardcoded string, put it in strings.xml it is better

File: app/src/main/res/layout/fragment_map.xml [hunk 7]
```diff
@@ -9,7 +9,67 @@
...
+            android:textSize="25sp" />
+
+        <TextView
+            android:id="@+id/fragment_map_event_time_end"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_margin="15dp"
+            app:layout_constraintTop_toBottomOf="@+id/fragment_map_event_time_start"
+            app:layout_constraintStart_toStartOf="parent"
+            android:text="End time"
```
Comment (TheImpressiveDonut) [inline app/src/main/res/layout/fragment_map.xml:58]: Hardcoded string, put it in strings.xml  it is better

File: app/src/main/res/layout/fragment_map.xml [hunk 8]
```diff
@@ -9,7 +9,67 @@
...
+            android:textSize="25sp" />
+
+        <TextView
+            android:id="@+id/fragment_map_event_location"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_margin="15dp"
+            app:layout_constraintTop_toBottomOf="@+id/fragment_map_event_time_end"
+            app:layout_constraintStart_toStartOf="parent"
+            android:text="Location"
```
Comment (TheImpressiveDonut) [inline app/src/main/res/layout/fragment_map.xml:68]: Hardcoded string, put it in strings.xml it is better

File: app/src/main/res/layout/fragment_map.xml [hunk 9]
```diff
@@ -9,7 +9,67 @@
...
-        android:layout_height="match_parent" />
+        android:layout_height="match_parent" >
 
+    </com.mapbox.mapboxsdk.maps.MapView>
+
+    <!--Button
+        android:id="@+id/fragment_map_test_show_bottom_sheet"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="Show collapsed bottom sheet" /-->
```
Comment (TheImpressiveDonut) [inline app/src/main/res/layout/fragment_map.xml:20]: Is this useful?

**Review by TheImpressiveDonut — CHANGES_REQUESTED** [review]
Nice code

**Review by TheImpressiveDonut — APPROVED** [review]

**Pull request merged**